sdk/core/azure-core/src/http/bearer_token_authentication_policy.cpp

```diff
@@ -18,10 +18,13 @@ std::unique_ptr<RawResponse> BearerTokenAuthenticationPolicy::Send(
   {
     std::lock_guard<std::mutex> lock(m_accessTokenMutex);
 
-    // Refresh the token in 2 or less minutes before the actual expiration.
-    if (std::chrono::system_clock::now() > (m_accessToken.ExpiresOn - std::chrono::minutes(2)))
+    if (std::chrono::system_clock::now()
+        > (m_accessToken.ExpiresOn - m_tokenRequestContext.MinimumExpiration))
     {
-      m_accessToken = m_credential->GetToken(m_tokenRequestContext, context);
+      auto tokenRequestContext = m_tokenRequestContext;
+      tokenRequestContext.MinimumExpiration += std::chrono::minutes(1);
```

Comment:
Why decrease the expiration by 1 minute here? Just trying to understand the algorithm.

Reply:
We're saying: "get me a new token, and if possible, get it so that is not expiring for at least next 3 minutes".

Default `MinimumExpiration` is 2 minutes, because it is used on line 22. So it comes to this:
"If token expires in 2 minutes or less" => ("get me a new token, and if possible, get it so that is not expiring for at least next 3 minutes")

Reply:
You know, thinking more about this, we don't actually need it to be 3 minutes. 2 minutes will work just fine.

Reply:
Yes, I updated it to 2 minutes. If check within `BearerTokenAuthPolicy` is going to see that token has <= 2 minutes to live, then by the time it gets for the `TokenCache` to decide whether the cached token is still good, the token will certainly have less lifetime when it had when `BearerTokenAuthPolicy` checked it.